src/com/projectkorra/projectkorra/firebending/FireShield.java

```diff
@@ -94,7 +94,8 @@ public void progress() {
 		if (!bPlayer.canBendIgnoreCooldowns(this)) {
 			remove();
 			return;
-		} else if ((!player.isSneaking() && shield) || (System.currentTimeMillis() > getStartTime() + shieldDuration && shield)) {
+		} else if ((!player.isSneaking() && shield) || (shieldDuration != 0 && System.currentTimeMillis() > getStartTime() + shieldDuration && shield)) {

```

Comment:
Have you noticed any side effects of using FireShield with a duration of 0? Is it infinite?

Reply:
Yeah it’s infinite now, haven’t noticed errors or side affects.

Reply:
What happens if it's negative?

Reply:
Idk